packages/vm/test/api/EIPs/eip-4200-static-relative.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { Chain, Common, Hardfork } from '@ethereumjs/common'
+import { FeeMarketEIP1559Transaction } from '@ethereumjs/tx'
+import { Address, privateToAddress } from '@ethereumjs/util'
+import * as tape from 'tape'
+
+import { VM } from '../../../src/vm'
+const pkey = Buffer.from('20'.repeat(32), 'hex')
+const GWEI = BigInt('1000000000')
+const sender = new Address(privateToAddress(pkey))
+
+async function runTx(vm: VM, data: string, nonce: number) {
+  const tx = FeeMarketEIP1559Transaction.fromTxData({
+    data,
+    gasLimit: 1000000,
+    maxFeePerGas: 7,
+    nonce,
+  }).sign(pkey)
+  const result = await vm.runTx({ tx })
+  const created = result.createdAddress
+  const code = created ? await vm.stateManager.getContractCode(created!) : undefined
+  return { result, code }
+}
+
+function getInt16Str(int16: number) {
+  const buf = Buffer.alloc(2)
+  buf.writeInt16BE(int16)
+  return buf.toString('hex')
+}
+
+function getRJUMPVCode(int16list: number[]) {
+  if (int16list.length > 255) {
+    throw new Error('does not fit')
+  }
+  let str = '5E' + int16list.length.toString(16).padStart(2, '0')
+  for (const i of int16list) {
+    str += getInt16Str(i)
+  }
+  return str
+}
+
+function getEOFCode(code: string) {
+  const magic = '0xEF000101'
+  const len = code.length / 2
+  const lenHex = len.toString(16)
+  return magic + lenHex.padStart(4, '0') + '00' + code
+}
+
+tape('EIP 3670 tests', (t) => {
+  const common = new Common({
+    chain: Chain.Mainnet,
+    hardfork: Hardfork.London,
+    eips: [3540, 3670, 4200],
```

Comment:
I created this PR before I found out about https://github.com/ethereum/EIPs/pull/6156, I think we can all merge it together and check for 3540 to see if EOF contracts are enabled or not (and not do the whole "is EIP `X` activated" thing)

Reply:
Ah, wait for it, sorry I was confused since this is only the *activation* of all these EIPs and nothing else.

Let's otherwise please handle the EIPs separately for now and not artificially "merge" into 3540 (if that's what you proposed), that would for sure confuse people and we would deviate from what the EIP contains.

I guess if we want a unifying solution here we would need to carefully discuss (since lightclient mentioned there will likely be non combinatory one-EIP-rule-them-all EIP number). We could "invent" our number and then allow for both (activating separately or by this invented number, hmm. Not sure if this is so optimal.

Reply:
To clarify, I did not mean to merge all of them into 3540, but at this point (as of https://github.com/ethereum/EIPs/blob/4e5cd9fe6dd168165a7a5cfef68ead439ef27f17/ this version of the EIPs), then if you activate 3540 you will activate all big EOF EIPs (if you do not do this then common will throw) - so internally in our code we can just check if 3540 is activated, then we implicitly also know that 4200 is activated.